spec/lib/importers/youtube_data_spec.rb

```diff
@@ -1,70 +1,207 @@
-require 'spec_helper'
+# frozen_string_literal: true
 
 describe YoutubeData do
   fixtures :affiliates, :rss_feeds, :rss_feed_urls, :youtube_profiles
 
   describe '.refresh' do
     let(:rss_feed) { mock_model RssFeed }
     let(:profile) { mock_model YoutubeProfile }
-    let(:youtube_data) { double YoutubeData }
+    let(:youtube_data) { double described_class }
+    let(:active_profiles) { double YoutubeProfile }
+    let(:refresh) do
+      t = Thread.new { described_class.refresh }
+      sleep 0.1
+      t.kill
+    end
 
     before do
-      allow(YoutubeProfile).to receive_message_chain(:active, :stale).and_return([profile], [])
-      expect(YoutubeData).to receive(:new).with(profile).and_return youtube_data
+      allow(YoutubeProfile).to receive(:active).and_return(active_profiles)
+      allow(active_profiles).to receive(:stale).and_return([profile], [])
+      allow(described_class).to receive(:new).
+        with(profile).
+        and_return(youtube_data)
+      allow(youtube_data).to receive(:import)
     end
 
-    it 'should import each profile' do
-      expect(youtube_data).to receive :import
+    it 'imports each profile' do
+      refresh
 
-      t = Thread.new { YoutubeData.refresh }
-      sleep 0.1
-      t.kill
+      expect(youtube_data).to have_received(:import)
+      expect(described_class).to have_received(:new).
+        with(profile)
+    end
+
+    context 'when we have already updated the maximum number of profiles for today' do
+      let(:maximum_profile_updates_per_day) do
+        Rails.configuration.youtube['maximum_profile_updates_per_day']
+      end
+
+      before do
+        allow(described_class).to receive(:number_of_profiles_updated_today).
+          and_return(maximum_profile_updates_per_day)
+      end
```

Comment:
I believe the code as written is significantly clearer. In particular, I don't like embedding an arbitrary number which may well change.

Reply:
This code has been simplified somewhat by the switch to using the MAXIMUM_PROFILE_UPDATES_PER_DAY constant, instead of the config file; are you OK with it now?

Reply:
My preference is still to be explicit, specifically _because_ it's "arbitrary number which may well change". Being explicit in the spec is what protects us from unintentional changes, such as someone accidentally changing `300` to `3000` in the code itself. That said, I leave it up to you.

Reply:
Changed.